tests/rag_answers/test_invalid_json_retry.py

```diff
@@ -0,0 +1,141 @@
+from collections.abc import Sequence
+from typing import Callable
+from unittest.mock import AsyncMock
+
+import pytest
+from deepeval.models.llms.amazon_bedrock_model import AmazonBedrockModel
+
+from govuk_chat_evaluation.rag_answers.invalid_json_retry import (
+    attach_invalid_json_retry_to_model,
+)
+
+
+INVALID_JSON_ERROR = (
+    "Evaluation LLM outputted an invalid JSON. Please use a better evaluation model."
+)
+
+
+@pytest.fixture
+def make_wrapped_model(mocker) -> Callable[..., tuple[AmazonBedrockModel, AsyncMock]]:
+    def _make(outcomes: Sequence[object], *, max_attempts: int = 3):
+        original = mocker.AsyncMock(side_effect=list(outcomes))
+        model = mocker.Mock(spec=AmazonBedrockModel)
+        model.a_generate = original
+        attach_invalid_json_retry_to_model(model, max_attempts=max_attempts)
+        return model, original
+
+    return _make
+
+
+class TestAttachInvalidJsonRetryToModel:
+    def test_attach_invalid_json_retry_to_model_requires_positive_attempts(
+        self, mocker
+    ):
+        model = mocker.Mock(spec=AmazonBedrockModel)
+        model.a_generate = mocker.AsyncMock()
+
+        with pytest.raises(ValueError, match="max_attempts must be a positive integer"):
+            attach_invalid_json_retry_to_model(model, max_attempts=0)
+
+        with pytest.raises(ValueError, match="max_attempts must be a positive integer"):
+            attach_invalid_json_retry_to_model(model, max_attempts=-5)
+
+    @pytest.mark.asyncio
+    async def test_attach_invalid_json_retry_to_model_wraps_a_generate(self, mocker):
+        original = mocker.AsyncMock(return_value=("ok", 0))
+        model = mocker.Mock(spec=AmazonBedrockModel)
+        model.a_generate = original
+
+        attach_invalid_json_retry_to_model(model, max_attempts=1)
+
+        assert model.a_generate is not original
+        result = await model.a_generate("prompt")
+        assert result == ("ok", 0)
+        original.assert_awaited_once()
+
+
+class TestMonkeyPatchedAGenerate:
```

Comment:
I think this should be a nested class of the above class as this is still part of the tests for attach_invalid_json_retry_to_model

Reply:
I've looked back at my previous comment and realised because I didn't use code blocks my indentation was lost - apologies for that. I've now put it in a code block to preserve that.